resources/Microsoft.VSCode.Dsc/Microsoft.VSCode.Dsc.psm1

```diff
@@ -90,9 +123,6 @@ function Invoke-VSCode
 }
 #endregion Functions
 
-# Keeps the path of the code.exe CLI path.
-$VSCodeCLIPath = Get-VSCodeCLIPath
-
```

Comment:
I think it might be better to add a boolean parameter like `$UseInsiders` instead of creating a completely new DSC resource. That way we don't have to maintain two resources that are essentially doing the same thing.

You can use the boolean parameter inside the static constructor to check if `$UseInsiders = $true` and set $VSCodeCLIPath with the correct insiders code.exe path. So basically:

```powershell
if $UseInsiders
{
    $VSCodeCLIPath = Get-VSCodeInsidersCLIPath
}
```

Reply:
@ryfu-msft Thanks for the review. This is exactly what I wanted to do in the first place, but couldn't grasp my head around it. My class-based PowerShell development is a bit frisky.

Wouldn't it require to add an additional property in the class, allowing the ability to choose between Insiders or Stable?

Reply:
Yup, you would need to add another boolean DSCProperty and use that to determine whether we point to the Insiders or Stable VSCodeCLI path.

Reply:
@ryfu-msft  Hey Ryan, I've rewritten the `Get-VSCodeCLIPath` (also started on the custom path one from #61 ). I'm struggling with how to get it initialized. Whenever `Initialize` is called, a loop occurs. I'm seeing something over the head. Do you mind taking a look?

Reply:
After struggling for some time now, I reconstructed the constructors. Because of the following logic when calling Invoke-DscResource:

![image](https://github.com/user-attachments/assets/d74eca25-b18a-43fc-b506-9a346f762c2f)

The call is made later to determine the CLI path. The export has an overload added. It's good to check if this is possible with `dsc.exe` I guess.